exporter/elasticsearchexporter/factory.go

```diff
@@ -206,16 +202,32 @@ func exporterhelperOptions(
 		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: false}),
 		exporterhelper.WithStart(start),
 		exporterhelper.WithShutdown(shutdown),
-		exporterhelper.WithQueue(cfg.QueueSettings),
 	}
+	qs := cfg.QueueSettings
 	if cfg.Batcher.enabledSet {
-		opts = append(opts, exporterhelper.WithBatcher(cfg.Batcher.BatcherConfig)) //nolint:staticcheck
+		if cfg.Batcher.Enabled {
+			qs.Batch = &exporterhelper.BatchConfig{
+				FlushTimeout: cfg.Batcher.FlushTimeout,
+				MinSize:      cfg.Batcher.MinSize,
+				MaxSize:      cfg.Batcher.MaxSize,
+				// TODO: Uncomment once core dependency updated.
+				// Sizer:        cfg.Batcher.Sizer,
+			}
+
+			// If the deprecated batcher is enabled without a queue, enable blocking queue to replicate the
+			// behavior of the deprecated batcher.
+			if !qs.Enabled {
+				qs.Enabled = true
+				qs.WaitForResult = true
+			}
+		}
 
 		// Effectively disable timeout_sender because timeout is enforced in bulk indexer.
 		//
 		// We keep timeout_sender enabled in the async mode (Batcher.Enabled == nil),
 		// to ensure sending data to the background workers will not block indefinitely.
 		opts = append(opts, exporterhelper.WithTimeout(exporterhelper.TimeoutConfig{Timeout: 0}))
 	}
+	opts = append(opts, exporterhelper.WithQueue(qs))
```

Comment:
if you do
```
	if err := qs.Validate(); err != nil {
		panic("invalid config " + err.Error())
	}
```

it is failing with

```
`batch` supports only `items` or `bytes` sizer`
```

which is what's failing integration tests. We did not validate our manually transformed config and this misconfiguration causes batcher to fail.